src/OneExplorer/OneStorage.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import {obtainWorkspaceRoot} from '../Utils/Helpers';
+import {Logger} from '../Utils/Logger';
+
+import {ConfigObj} from './ConfigObject';
+
+interface StringMap {
+  [key: string]: string[];
+}
+
+interface ConfigObjMap {
+  [key: string]: ConfigObj|null;
+}
+
+/**
+ * A singleton storage class
+ *
+ * PURPOSE
+ *
+ * To build each 'Node' of OneTreeDataProvider,
+ * it is neccessary to access the file system, read the files and build objects(ConfigObj, ...).
+ * By keeping some file system information as data structure (list, map),
+ * some duplicated works can be reduced.
+ *
+ * LIFE CYCLE
+ *
+ * The singleton is created when the first get() is called.
+ * The object remains until OneStorage.reset() is called.
+ * OneStorage.reset() is called within OneTreeDataProvider.refresh(), which is called on every file system
+ * change within the repository.
+ */
+export class OneStorage {
```

Comment:
This class seems available for unit test but not sure... is it available or not? @.@

Reply:
Seemingly, it's possible to add some unit tests.
I will add them later on :-D